combine components in main.py

Theskrtnerd opened the pull request.

Theskrtnerd committed:
combine components in main.py

thaohp113 commented:
## Check requirement

- **Design/plan**: currently there’s no clear overview of design/plan documentation → ***Next step***: must add design/plan (ex: which techstacks have been used, tasks,… - adding either design or plan is fine I guess)
- **Testing:** Unit test is not compulsory, but since your team has written some, here are some comments:
    - Run poetry run pytest return an error at file tests/test_file_handler.py at test case 2 because it’s not compatible with linux → ***Next step:*** should handle this test case for Linus, although it has passed the CI test
    - Test image is not suitable since there’s no text on the image

<img width="1369" alt="Screenshot 2024-07-04 at 10 23 32 AM" src="https://github.com/Theskrtnerd/cinnamon-ai-bootcamp-w1/assets/142286712/95dea122-aac2-4d6f-a0b2-8b7ea399d461"> 

- **Output:** The last 2 pull requests have not been merged, and no reviewer has been added yet. → ***Next step:*** review the team’s work flow and merge these two PRs into main branch
    
    
<img width="1369" alt="Screenshot 2024-07-04 at 10 25 17 AM" src="https://github.com/Theskrtnerd/cinnamon-ai-bootcamp-w1/assets/142286712/f33f445c-7c1b-4e6c-940b-3570fac973f5">

## Code convention

- **Documentation**: there’s no clear and consistent documentation for code - no docstrings, some functions have comments, but some other don’t (the reference is from `file_handler.py` → ***Next step***: fix documentation and add comment
<img width="974" alt="Screenshot 2024-07-04 at 10 25 48 AM" src="https://github.com/Theskrtnerd/cinnamon-ai-bootcamp-w1/assets/142286712/35b72503-8a14-4f8d-8df0-171f2d5734bc">

- **Some part is not clean yet**: some codes are currently commented out at `ocr_parser.py` - which is quite confusing (guess this is because your team has yet finalized the code) → ***Next step***: finalize and clean these codes.
    
<img width="1362" alt="Screenshot 2024-07-04 at 10 27 12 AM" src="https://github.com/Theskrtnerd/cinnamon-ai-bootcamp-w1/assets/142286712/1ddc328d-e92a-46d1-811a-9188c9c3bcd4">

hungpham3112 reviewed (changes requested):
You should add two more flags in CLI  for specific input and output instead of forcing users to use /input and /output.

Theskrtnerd committed:
make changes to pr reviews

Theskrtnerd committed:
make changes to pr reviews

Theskrtnerd committed:
fix test files, add design plan

Theskrtnerd commented:
> ## Check requirement
> * **Design/plan**: currently there’s no clear overview of design/plan documentation → _**Next step**_: must add design/plan (ex: which techstacks have been used, tasks,… - adding either design or plan is fine I guess)
> * **Testing:** Unit test is not compulsory, but since your team has written some, here are some comments:
>   
>   * Run poetry run pytest return an error at file tests/test_file_handler.py at test case 2 because it’s not compatible with linux → _**Next step:**_ should handle this test case for Linus, although it has passed the CI test
>   * Test image is not suitable since there’s no text on the image
>   * Some test files are blank
> 
> <img alt="Screenshot 2024-07-04 at 10 23 32 AM" width="1369" src="https://private-user-images.githubusercontent.com/142286712/345686408-95dea122-aac2-4d6f-a0b2-8b7ea399d461.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAwODE0MjcsIm5iZiI6MTcyMDA4MTEyNywicGF0aCI6Ii8xNDIyODY3MTIvMzQ1Njg2NDA4LTk1ZGVhMTIyLWFhYzItNGQ2Zi1hMGIyLThiN2VhMzk5ZDQ2MS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcwNFQwODE4NDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xMjk1YzYxY2FjYzFiNGIzMmNlMzRmMGZhNjVjZDA3Mzg4MzdkNWU1M2I2ZmUyZGE0NjkyZmI1ODcwZTQyZGE4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.tF0QwX9ZtQxB_XkkKT64aHG4-ON9hvmbLDTD4JuElKo">
> * **Output:** The last 2 pull requests have not been merged, and no reviewer has been added yet. → _**Next step:**_ review the team’s work flow and merge these two PRs into main branch
> 
> <img alt="Screenshot 2024-07-04 at 10 25 17 AM" width="1369" src="https://private-user-images.githubusercontent.com/142286712/345686693-f33f445c-7c1b-4e6c-940b-3570fac973f5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAwODE0MjcsIm5iZiI6MTcyMDA4MTEyNywicGF0aCI6Ii8xNDIyODY3MTIvMzQ1Njg2NjkzLWYzM2Y0NDVjLTdjMWItNGU2Yy05NDBiLTM1NzBmYWM5NzNmNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcwNFQwODE4NDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00Yjg2YTA4ZTA4NmM2NmFkMjU3MzQzMjMxNGNkNjQ0ZGMzMWU4ZGNmOGIzZTE5OTRiN2FmODYzMDYzYjkyNTU1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.pFLhkfqBLsuJTIcxoI1Mg4u9Hcjs8C_BD60qzYl4Mos">
> ## Code convention
> * **Documentation**: there’s no clear and consistent documentation for code - no docstrings, some functions have comments, but some other don’t (the reference is from `file_handler.py` → _**Next step**_: fix documentation and add comment
> 
> <img alt="Screenshot 2024-07-04 at 10 25 48 AM" width="974" src="https://private-user-images.githubusercontent.com/142286712/345686801-35b72503-8a14-4f8d-8df0-171f2d5734bc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAwODE0MjcsIm5iZiI6MTcyMDA4MTEyNywicGF0aCI6Ii8xNDIyODY3MTIvMzQ1Njg2ODAxLTM1YjcyNTAzLThhMTQtNGY4ZC04ZGYwLTE3MWYyZDU3MzRiYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcwNFQwODE4NDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00ODc2MDk4MTEwYjMyZGM1ODYwM2M3NDhkZWU2MWM2YjdkNTI2M2U0YTcwYjVjOWE0Mzk2ODkzZGI4YTUzM2VmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Wc5-uuzcgngCRC9mMxzB2-4vios-JmTkdnMWeHNciFU">
> * **Some part is not clean yet**: some codes are currently commented out at `ocr_parser.py` - which is quite confusing (guess this is because your team has yet finalized the code) → _**Next step**_: finalize and clean these codes.
> 
> <img alt="Screenshot 2024-07-04 at 10 27 12 AM" width="1362" src="https://private-user-images.githubusercontent.com/142286712/345686991-1ddc328d-e92a-46d1-811a-9188c9c3bcd4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAwODE0MjcsIm5iZiI6MTcyMDA4MTEyNywicGF0aCI6Ii8xNDIyODY3MTIvMzQ1Njg2OTkxLTFkZGMzMjhkLWU5MmEtNDZkMS04MTFhLTkxODhjOWMzYmNkNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcwNFQwODE4NDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01NDhkZDdhZTdiM2M2ZDE5NTRkZDIzN2I2OGFiNDA3Mjk1MmQzM2I2ZDdmNjQwYTk2NTA2M2M5OGRhMzhlYWE5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.epVl80dhY0HYA9NoTfiRFkqg4Vy9zZVnhiLqSWm3anU">

Thank you so much for your detailed comment. For your suggestions, I have done the following:

* Design/plan: I've added our design plan in the repo (`design-plan.png`)
* Testing:
   * For the Linux test case, we will try to think of a different solution as our current Python module does not support that OS.
   * I've changed the test image to another one with text in it.
   * We will add the other test files later.
* Output: The 2 open PRs are meant to be reviewed before merging, and I have requested your team for review. Please give at least 2 reviews in order to merge the PR.
* Code convention: We will add the comments and clean the codebase later
*

hungpham3112 commented on llamagon_ocr_script/main.py:
```diff
@@ -32,7 +30,13 @@ def main(file_name, upload):
+    )
+    parser.add_argument(
+        "--output_folder", type=str, default="output", help="The folder to save the output images and text."
+    )
```
where is the file_name parser? You used `args.file_name` but didn't have an argument for it.
